Gems/PhysX/Code/Source/RigidBodyComponent.cpp

```diff
@@ -181,14 +188,19 @@ namespace PhysX
         if (m_cachedSceneInterface)
         {
             m_cachedSceneInterface->RemoveSimulatedBody(m_attachedSceneHandle, m_rigidBodyHandle);
+            m_rigidBodyHandle = AzPhysics::InvalidSimulatedBodyHandle;
         }
 
+        AZ::EntityBus::Handler::BusDisconnect();
         Physics::RigidBodyRequestBus::Handler::BusDisconnect();
         AzPhysics::SimulatedBodyComponentRequestsBus::Handler::BusDisconnect();
         AZ::TransformNotificationBus::MultiHandler::BusDisconnect();
         m_sceneFinishSimHandler.Disconnect();
         m_activeBodySyncTransformHandler.Disconnect();
         AZ::TickBus::Handler::BusDisconnect();
+
+        m_isLastMovementFromKinematicSource = false;
+        m_rigidBodyTransformNeedsUpdateOnPhysReEnable = false;
```

Comment:
nit: Can we rename `Phys` to `Physics` to avoid abbreviations that make names harder to understand

Reply:
Since this work is quite large already I will try not to do unnecessary clean-up changes to keep it manageable and focused.

Reply:
Okay fair enough